src/main/java/io/aiven/kafka/connect/http/recordsender/SingleRecordSender.java

```diff
@@ -32,13 +32,13 @@ protected SingleRecordSender(final HttpSender httpSender) {
     public void send(final Collection<SinkRecord> records) {
         for (final SinkRecord record : records) {
             final String body = recordValueConverter.convert(record);
-            httpSender.send(body);
+            httpSender.send(body, record.key().toString());
```

Comment:
Consider having a `recordKeyConverter` -- it should only convert known types (e.g. String, Numbers) into String to avoid messing with complex/unexpected conversions.

Reply:
Also, what happens when the key is null? Should it use an empty string? fail? use a fallback value?